packages/dart_frog_cli/lib/src/commands/dev/dev.dart

```diff
@@ -152,8 +153,10 @@ class DevCommand extends DartFrogCommand {
     }
 
     final watcher = _directoryWatcher(path.join(cwd.path));
-    final subscription =
-        watcher.events.where(shouldRunCodegen).listen((_) => codegen());
+    final subscription = watcher.events
+        .where(shouldRunCodegen)
+        .debounce(Duration.zero)
```

Comment:
This will prevent duplicate codegen executions when a file is renamed for example because a rename is treated as two back-to-back watcher events (REMOVE, ADD).

Reply:
Is that the case on all platforms? Doubt that this change would be an issue on non UNIX platforms but might be worth to confirm?

Reply:
Yeah I believe so but I'll confirm when I have access to a windows machine 👍